lib/versioning/utils.js

```diff
@@ -0,0 +1,36 @@
+const VSConst = require('./constants').VersioningConstants;
+
+const MSTLEN = VSConst.VERSIONING_AFFIXES.MASTER_VERSION_PREFIX.length;
+const VERLEN = VSConst.VERSIONING_AFFIXES.OTHER_VERSION_PREFIX.length;
+const SEP = VSConst.VERSIONING_AFFIXES.SEPARATOR;
+
+module.exports.VersioningUtils = {
+    // using explicit encode and decode functions for version object to enable
+    // future extensions, such as attatching external information to the string
+    // value from request instead of having to parse and stringify the value
+    // Example: "prefixlen_{prefix}{value_from_request}"
+    encodeVersion: version => JSON.stringify(version),
+    decodeVersion: version => JSON.parse(version),
+    // using these functions to manipulate the attributes of the value also
+    // help us from restricting ourselves to hardcoding arbitrary attributes
+    setattr: (version, attr, value) => version[attr] = value,
+    setts: (version, ts) => version[VSConst.ATTR_VERSION_ID] = ts,
+    getts: version => version[VSConst.ATTR_VERSION_ID],
```

Comment:
We used the term "timestamp" for unique identifier that increases monotonically and that happened to be used for "versionId". Though they are the same and "ts" is shorter but we can change it to "versionId" without any issue.